### ISSUE TITLE
Add an option to keep retrying even if first sync attempt fails

### ISSUE THREAD

**Issue by filmil** [kubernetes/git-sync#161]
I'd like to have a settable option by which `git-sync` will not abort on a failed first `git sync` attempt, but would keep retrying.

This seems a bit counter-intuitive.  I've found, however, that restarting a git-sync deployment that is seeing a transient error is less confusing than a git-sync that has reached a full 5 minute backoff delay and then gets fixed -- because if you are not careful you may miss the fact that you are seeing a backed-off git-sync.

In practice, we end up power-cycling git-sync anyways.  At which point, we may as well keep git-sync up and running.

Long story short, PR upcoming.
